src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/DriverFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\DependencyInjection\Driver;
+
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineODMDriver;
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineORMDriver;
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrinePHPCRDriver;
+use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Exception\UnknownDriverException;
+use Sylius\Bundle\ResourceBundle\SyliusResourceBundle;
+use Sylius\Component\Resource\Metadata\MetadataInterface;
+
+/**
+ * @author Arnaud Langlade <aRn0D.dev@gmail.com>
+ */
+class DriverFactory
+{
+    /**
+     * @var DriverInterface[]
+     */
+    static private $drivers = array();
+
+    /**
+     * @param MetadataInterface $metadata
+     *
+     * @return DriverInterface
+     *
+     * @throws UnknownDriverException
+     */
+    public static function get(MetadataInterface $metadata)
+    {
+        $type = $metadata->getDriver();
+
+        if (isset(self::$drivers[$type])) {
+            return self::$drivers[$type];
```

Comment:
It's quite counterintuitive that `DriverFactory` does not always create new driver.